Clean up some uses of `String::substr`

AThousandShips opened the pull request:
Cases where the end position is either equvalent to the default or past the end of the string.

Cases:
* `X.substr(Y, X.length())` (which is always equivalent or beyond the end)
* `X.substr(Y, X.length() - Y)`, `X.substr(Y, X.size() - (Y + 1))` (which are always equivalent)
* `X.substr(Y, X.size())` or `X.substr(Y, X.size() - Y)` (which is always beyond the end)
* `X.substr(Y, -1)` (which is the default)

For context, `X.substr(Y, -1)` is equivalent to `X.substr(Y, X.length() - Y)`, and if the second argument is greater than `X.length() - Y` it is clamped, so equivalent to the default again. Note that `X.size()` is always `X.length() + 1` unless `X.length() == 0` (in which case `substr` returns empty)

This communicates this intent more clearly, and avoids unnecessary operations, and avoids errors due to copying the code or minor changes down the line.

Didn't try to identify cases equivalent to this where the second argument was a variable.

<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

Ivorforce reviewed (approved):
I skimmed all changes; looks correct to me. I agree it's cleaner without an explicit second argument.

I would recommend waiting to merge until 4.5 due to the non zero risk of simply having gotten one of those wrong.

AThousandShips commented:
Absolutely for no sooner than 4.5, went over them with a few specific patterns (including some negations like `X.substr(A + B, X.size() - A - B)` and similar which were a bit harder to detect

Can split into individual categories for ease of review (i.e. the trivial case of `X.substr(Y, X.length())`, but shouldn't be difficult to work through the validity of them

bruvzg reviewed (approved):
Looks good.

AThousandShips committed:
Clean up some uses of `String::substr`

Cases where the end position is either equvalent to the default or past
the end of the string.

Repiteo commented:
Thanks!

AThousandShips commented:
Thank *you*!
